packages/react-experiments/src/components/SelectedItemsList/Items/subcomponents/DefaultEditingItem.tsx

```diff
@@ -41,118 +45,215 @@ export interface IDefaultEditingItemInnerProps<TItem> extends React.HTMLAttribut
+        onFloatingSuggestionsDismiss={props.onDismiss}
+        suggestions={suggestionItems}
+        selectedSuggestionIndex={focusItemIndex}
+        selectedHeaderIndex={headerItemIndex}
+        selectedFooterIndex={footerItemIndex}
+        pickerSuggestionsProps={pickerSuggestionsProps}
       />
     );
   };
 
-  private _resolveInputRef = (ref: HTMLInputElement): void => {
-    this._editingInput = ref;
+  const _onInputBlur = React.useCallback(
+    (ev: React.FocusEvent<HTMLElement>): void => {
+      if (focusItemIndex >= 0) {
+        _onSuggestionSelected(ev, suggestionItems[focusItemIndex]);
+      } else if (createGenericItem) {
+        onEditingComplete(item, createGenericItem(inputValue));
+      }
+      // else, we come out of editing mode
+    },
+    // eslint-disable-next-line react-hooks/exhaustive-deps -- these are the only dependencies which matter
+    [suggestionItems, focusItemIndex, item, createGenericItem, inputValue, onEditingComplete],
+  );
 
-    this.forceUpdate(() => {
-      this._editingInput.focus();
-    });
-  };
+  const _onInputChange = React.useCallback(
+    (ev: React.FormEvent<HTMLElement>): void => {
+      const value: string = (ev.target as HTMLInputElement).value;
+      setInputValue(value);
 
-  private _onInputClick = (): void => {
-    this._editingFloatingPicker.current && this._editingFloatingPicker.current.showPicker(true /*updatevalue*/);
-  };
+      if (value === '') {
+        if (onRemoveItem) {
+          onRemoveItem(item);
+        }
+      } else {
+        setEditingSuggestions(getSuggestions(value));
+      }
+    },
+    [onRemoveItem, getSuggestions, item],
+  );
 
-  private _onInputBlur = (ev: React.FocusEvent<HTMLElement>): void => {
-    if (this._editingFloatingPicker.current && ev.relatedTarget !== null) {
-      const target = ev.relatedTarget as HTMLElement;
-      if (
-        target.className.indexOf('ms-Suggestions-itemButton') === -1 &&
-        target.className.indexOf('ms-Suggestions-sectionButton') === -1
-      ) {
-        this._editingFloatingPicker.current.forceResolveSuggestion();
+  const _onInputKeyDown = React.useCallback(
+    (ev: React.KeyboardEvent<HTMLInputElement>): void => {
+      const keyCode = ev.which;
+      switch (keyCode) {
+        case KeyCodes.backspace:
+        case KeyCodes.del:
+          ev.stopPropagation();
+          break;
+        case KeyCodes.enter:
+        case KeyCodes.tab:
+          if (!ev.shiftKey && !ev.ctrlKey && (focusItemIndex >= 0 || footerItemIndex >= 0 || headerItemIndex >= 0)) {
+            ev.preventDefault();
+            ev.stopPropagation();
+            if (focusItemIndex >= 0) {
+              // Get the focused element and add it to selectedItemsList
+              _onSuggestionSelected(ev, suggestionItems[focusItemIndex]);
+            } else if (footerItemIndex >= 0) {
+              // execute the footer action
+              footerItems![footerItemIndex].onExecute!();
```

Comment:
just double checking that there's no way `footerItems` and `onExecute` can be undefined, and you don't want to use `?.` instead of `!`?

Reply:
footerItemIndex will be -1 if those aren't defined, so we'll never get to this line